File: Permacore/backend/routes/rejection_code_routes.py
from flask import Blueprint, request, jsonify
from ..services.rejection_code_service import RejectionCodeService

rejection_code_bp = Blueprint('rejection_code_bp', __name__)

@rejection_code_bp.route('/api/rejection_codes', methods=['POST'])
def add_rejection_code():
    data = request.json
    rejection_code = RejectionCodeService.create_rejection_code(data)
    return jsonify(rejection_code), 201

@rejection_code_bp.route('/api/rejection_codes', methods=['GET'])
def get_rejection_codes():
    rejection_codes = RejectionCodeService.get_all_rejection_codes()
    return jsonify([code.__dict__ for code in rejection_codes]), 200

@rejection_code_bp.route('/api/rejection_codes/<int:code_id>', methods=['PUT'])
def update_rejection_code(code_id):
    data = request.json
    rejection_code = RejectionCodeService.update_rejection_code(code_id, data)
    if rejection_code:
        return jsonify(rejection_code), 200
    return jsonify({"message": "Rejection code not found"}), 404

@rejection_code_bp.route('/api/rejection_codes/<int:code_id>', methods=['DELETE'])
def delete_rejection_code(code_id):
    if RejectionCodeService.delete_rejection_code(code_id):
        return jsonify({"message": "Rejection code deleted"}), 200
    return jsonify({"message": "Rejection code not found"}), 404

File: MERN_Permacore/server/src/api/routes/rejectionCodeRoutes.js
const express = require('express');
const router = express.Router();

// Import the rejection code controller
const rejectionCodeController = require('../controllers/rejectionCodeController');

// GET all rejection codes
router.get('/', rejectionCodeController.getAllRejectionCodes);

// GET a single rejection code by id
router.get('/:id', rejectionCodeController.getRejectionCodeById);

// POST a new rejection code
router.post('/', rejectionCodeController.createRejectionCode);

router.post('/check-duplicate', async (req, res) => {
    try {
        const { codeId, codeNumber, category } = req.body;

        // Validate input
        if (!codeId || !codeNumber || !category) {
            return res.status(400).json({ message: 'Code ID, Code Number, and Category are required.' });
        }

        const existingCode = await RejectionCode.findOne({ codeId, codeNumber, category });
        res.status(200).json({ exists: !!existingCode });
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
});


// PUT to update a rejection code
router.put('/:id', rejectionCodeController.updateRejectionCode);

// DELETE a rejection code
router.delete('/:id', rejectionCodeController.deleteRejectionCode);

module.exports = router;
